tests/resource_manager.rs

```diff
@@ -0,0 +1,280 @@
+
+        fn send_request_with_reply<R>(
+            &self,
+            _: &[IoSlice<'_>],
+            _: Vec<RawFdContainer>,
+        ) -> Result<Cookie<'_, Self, R>, ConnectionError>
+        where
+            R: for<'a> TryFrom<&'a [u8], Error = ParseError>,
+        {
+            Ok(Cookie::new(self, 42))
+        }
+
+        fn send_request_with_reply_with_fds<R>(
+            &self,
+            _: &[IoSlice<'_>],
+            _: Vec<RawFdContainer>,
+        ) -> Result<CookieWithFds<'_, Self, R>, ConnectionError>
+        where
+            R: for<'a> TryFrom<(&'a [u8], Vec<RawFdContainer>), Error = ParseError>,
+        {
+            todo!()
+        }
+
+        fn send_request_without_reply(
+            &self,
+            _: &[IoSlice<'_>],
+            _: Vec<RawFdContainer>,
+        ) -> Result<VoidCookie<'_, Self>, ConnectionError> {
+            todo!()
+        }
+
+        fn discard_reply(&self, _: SequenceNumber, _: RequestKind, _: DiscardMode) {
+            todo!()
+        }
+
+        fn prefetch_extension_information(&self, _: &'static str) -> Result<(), ConnectionError> {
+            todo!()
+        }
+
+        fn extension_information(
+            &self,
+            _: &'static str,
+        ) -> Result<Option<ExtensionInformation>, ConnectionError> {
+            todo!()
+        }
+
+        fn wait_for_reply_or_raw_error(
+            &self,
+            sequence: SequenceNumber,
+        ) -> Result<ReplyOrError<Self::Buf>, ConnectionError> {
+            let value = self.1.as_ref().map(|v| &v[..]).unwrap_or(&[]);
+            let mut reply = Vec::new();
+            // response type
+            reply.push(1);
+            // format
+            if value.is_empty() {
+                reply.push(0);
+            } else {
+                reply.push(8);
+            }
+            // sequence
+            (sequence as u16).serialize_into(&mut reply);
+            // length
+            0u32.serialize_into(&mut reply);
+            // type (STRING)
+            31u32.serialize_into(&mut reply);
+            // bytes_after
+            0u32.serialize_into(&mut reply);
+            (value.len() as u32).serialize_into(&mut reply);
+            // padding
+            reply.extend([0; 12].iter().copied());
+            reply.extend(value);
+            Ok(ReplyOrError::Reply(reply))
+        }
+
+        fn wait_for_reply(&self, _: SequenceNumber) -> Result<Option<Self::Buf>, ConnectionError> {
+            todo!()
+        }
+
+        fn wait_for_reply_with_fds_raw(
+            &self,
+            _: SequenceNumber,
+        ) -> Result<ReplyOrError<BufWithFds<Self::Buf>, Self::Buf>, ConnectionError> {
+            todo!()
+        }
+
+        fn check_for_raw_error(
+            &self,
+            _: SequenceNumber,
+        ) -> Result<Option<Self::Buf>, ConnectionError> {
+            todo!()
+        }
+
+        fn prefetch_maximum_request_bytes(&self) {
+            todo!()
+        }
```

Comment:
Are those `todo!()`s expected to be implemented or left as is?

Reply:
Since this is just in a test: I would just leave them. If something in the code changes so that the test now needs them, then they can be implemented.

This is just all an elaborate scheme so that I can mock `get_property` and return the expected data. Perhaps the code here might need some more comments...

Reply:
Then I would use `unimplemented!()` instead of `todo!()`  (see [this](https://doc.rust-lang.org/stable/std/macro.todo.html)).

Reply:
You are right. I pushed a new commit that does that.